azkaban-exec-server/src/test/java/azkaban/execapp/JobRunnerTest.java

```diff
@@ -286,9 +286,11 @@ public void testDelayedExecutionCancelledJob() throws InterruptedException {
     final Thread thread = new Thread(runner);
     thread.start();
 
-    Thread.sleep(2000);
+    waitForStatus(node, Status.READY);
+    // sleep so that job has time to get into delayExecution() -> wait()
+    Thread.sleep(1000L);
```

Comment:
This could also be the previous 2 seconds and offer maybe better stability, but on the other hand it will make the test run take longer than before, because below we need to add a new 1 s sleep.